src/kernels/internalTypes.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { CancellationToken } from 'vscode';
+import { Resource } from '../platform/common/types';
+import { KernelConnectionMetadata } from './types';
+
+export interface IContributedKernelFinder {

```

Comment:
Can we call these `IKernelFinder` or the like, feels weierd to call them `Contributed` after all everything will be contributed in the extension, as we're creating implementations of interafces, also everything is in this extension.
If its something that can be contributed by another extension, then i think thats fine

Reply:
Not, necessary to change, but nit pick

Reply:
`IKernelFinder` is public to external components while `ContributedKernelFinder` are the raw/remote kernel finder impls and they are internal details (not exposed to external modules). Called it `contributed*` as they are contributions/plugins to the main `IKernelFinder`. I'm imagining it can be contributed by other extensions (say we want to have the Python envs from Python extension), with that said, I'm fine with calling it something like `IKernelFinderImpl`.

Reply:
no no, lets leave this.
Its just that when I think of `contribution`, to me thats basically `external` extensibility, or `external` contribution.
But i'm fine with this.